File: script/load_instance.rb
require 'json'

Root = "#{File.dirname(__FILE__)}/../"

class Instance
  def initialize hash
    @hash = hash
  end
  def name
    @hash["name"]
  end
  def jobs
    @hash["jobs"]
  end
  def machines
    @hash["machines"]
  end
  def operations
    self.jobs * self.machines
  end
  def path
    @hash["path"]
  end
  def optimum
    opt = @hash["optimum"]
    bnd = @hash["bounds"]
    return "nan" if [opt,bnd].compact.empty?
    return opt unless opt.nil?
    bnd["lower"]
  end
  def bounds
    @hash["bounds"]
  end
end


def load_instance target, dir = Root
  file = open("#{Root}/instances.json")
  json = JSON.parse( file.readlines.map(&:chomp).join )
  instance = json.select{|data| data["name"] == target }
  raise "There is no instance named '#{target}'" if instance.empty?
  instance = instance.first
  instance["path"] = File.expand_path( instance["path"], dir )
  Instance.new( instance )
end

if $0 == __FILE__
  target,dir = ARGV
  dir ||= Root
  instance = load_instance( target, dir )
  print "--instance %s --optimum %s" % [instance.path,instance.optimum]
end

File: script/load_instance.py
# coding: utf-8

import os
import sys
import json
import math

Argv = sys.argv
if( len(Argv) < 2 ):
    sys.stderr.write( "Usage: python %s <instance>\n" % Argv[0] )
    quit()

Target = Argv[1]

file = open( os.path.join( os.path.dirname( __file__ ), 'instances.json' ), "r" )
data = json.load( file )

instance = [ inst for inst in data if inst['name'] == Target ]
if( len(instance) == 0 ):
    raise Exception("There is no instance named %s" % Target)

instance = instance[0]
path = os.path.abspath( os.path.join(os.path.dirname( __file__ ),instance['path']) )
optimum = instance['optimum']
print( optimum )
if( optimum is None ):
    if( instance['bounds'] is None ):
        optimum = "nan"
    else:
        optimum = instance['bounds']['lower']

sys.stdout.write('--instance %s --optimum %s' % (path,optimum))
